### PR TITLE
dlb: upgrade to 8.8

### PR DESCRIPTION
**Pull request by daixiang0**
Commit Message: Upgrade libdlb to 8.8
Additional Description:
Risk Level: Low
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by daixiang0** [comment]
/retest

download jq timeout, https://github.com/envoyproxy/envoy/actions/runs/8864305286

**Comment by daixiang0** [comment]
Now the latest layout is like:
```
dlb
├── driver
│   └── dlb2
│       ├── user.h
└── libdlb
    ├── cli
    │   ├── dlb_monitor_sec.c
    │   ├── dump_dlb2_5_regs.h
    │   ├── dump_dlb_regs.c
    │   ├── dump_dlb_regs.h
    │   ├── Makefile
    │   └── open_dlb.c
    ├── dlb2_ioctl.c
    ├── dlb2_ioctl.h
    ├── dlb_adv.h
    ├── dlb.c
    ├── dlb_common.h
    ├── dlb.h
    ├── dlb_ops.h
    ├── dlb_priv.h
    ├── doc
    │   ├── Doxyfile
    │   └── Makefile
    ├── examples
    │   ├── dir_traffic.c
    │   ├── example_buffer.h
    │   ├── ldb_traffic.c
    │   └── Makefile
    ├── license
    │   ├── BSD-3-Clause
    │   ├── GPL-2.0
    │   └── Linux-syscall-note
    ├── Makefile
    └── version.map
```

I want to run `make` under `libdlb` dir and it depends on `driver/dlb2/user.h`.

I can not use filegroup() to include the `user.h` since it would make bazel cannot find Makefile.

@phlax is there a way to do it?

File: bazel/repositories_extra.bzl [hunk 1]
```diff
@@ -15,7 +15,7 @@ PYTHON_MINOR_VERSION = _python_minor_version(PYTHON_VERSION)
 # Envoy deps that rely on a first stage of dependency loading in envoy_dependencies().
 def envoy_dependencies_extra(
         python_version = PYTHON_VERSION,
-        ignore_root_user_error = False):
+        ignore_root_user_error = True):
```
Comment (adisuissa) [inline bazel/repositories_extra.bzl:18]: cc @phlax
I think we discussed this a while ago (maybe I'm misremembering).

@daixiang0 why is this needed?
Can this be added to a specific dependency, and not impact all dependencies?

File: contrib/dlb/source/BUILD [hunk 2]
```diff
@@ -15,6 +15,7 @@ envoy_contrib_package()
 
 make(
     name = "dlb",
+    env = {"DLB_DISABLE_DOMAIN_SERVER": "TRUE"},
```
Comment (adisuissa) [inline contrib/dlb/source/BUILD:18]: I think this change already appeared in another PR.

File: contrib/contrib_build_config.bzl [hunk 3]
```diff
@@ -4,58 +4,6 @@ CONTRIB_EXTENSIONS = {
     # Compression
     #
 
-    "envoy.compression.qatzip.compressor":                      "//contrib/qat/compression/qatzip/compressor/source:config",
```
Comment (adisuissa) [inline contrib/contrib_build_config.bzl:7]: Why are these extensions removed?

**Review by adisuissa — COMMENTED** [review]
/wait

File: bazel/repositories_extra.bzl (hunk 1 above)
Comment (phlax) [reply bazel/repositories_extra.bzl:18]: yeah - we dont want this

Comment (phlax) [reply bazel/repositories_extra.bzl:18]: i dont think this is related to the PR tbh - its related to local setup

File: ci/do_ci.sh [hunk 4]
```diff
@@ -563,10 +563,6 @@ case $CI_TARGET in
         echo "bazel fastbuild build with contrib extensions and tests..."
         echo "Building..."
         bazel_contrib_binary_build fastbuild
-        echo "Testing ${TEST_TARGETS[*]}"
-        bazel test "${BAZEL_BUILD_OPTIONS[@]}" \
-              -c fastbuild \
-              "${TEST_TARGETS[@]}"
```
Comment (phlax) [inline ci/do_ci.sh:569]: this doesnt look right

**Comment by daixiang0** [comment]
@phlax @adisuissa sorry that I add some local changes here :(

**Comment by daixiang0** [comment]
I try to move files into one dir by Bazel and fail, so I just move the file by patch.

File: contrib/dlb/source/BUILD [hunk 5]
```diff
@@ -16,7 +16,6 @@ envoy_contrib_package()
 make(
     name = "dlb",
     env = {"DLB_DISABLE_DOMAIN_SERVER": "TRUE"},
-    includes = [],
```
Comment (phlax) [inline contrib/dlb/source/BUILD:19]: rather than patching the whole file in could you do as i suggested offline, add a filegroup for the includes directory, and then add it here - ie `@intel_dlb//:libdlb_includes`

Comment (daixiang0) [reply contrib/dlb/source/BUILD:19]: Yeah, I tried it but fail:
```
--- a/bazel/repositories.bzl
+++ b/bazel/repositories.bzl
@@ -1370,6 +1370,12 @@ def _intel_dlb():
     external_http_archive(
         name = "intel_dlb",
         build_file_content = """
+filegroup(
+    name = "libdlb_header",
+    srcs = glob(["**/dlb2_user.h"]),
+    visibility = ["@envoy//contrib/dlb/source:__pkg__"],
+)
+
 filegroup(
     name = "libdlb",
     srcs = glob(["dlb/libdlb/*"]),
diff --git a/contrib/dlb/source/BUILD b/contrib/dlb/source/BUILD
index 46c28a3dad..98c12bf526 100644
--- a/contrib/dlb/source/BUILD
+++ b/contrib/dlb/source/BUILD
@@ -16,6 +16,7 @@ envoy_contrib_package()
 make(
     name = "dlb",
     env = {"DLB_DISABLE_DOMAIN_SERVER": "TRUE"},
+   includes = ["@intel_dlb//:libdlb_header"]
     lib_source = "@intel_dlb//:libdlb",
     out_static_libs = ["libdlb.a"],
     postfix_script = "mv libdlb.a $INSTALLDIR/lib && rm -rf $INSTALLDIR/include && mkdir -p $INSTALLDIR/include && cp -L *.h $INSTALLDIR/include",
```

Comment (daixiang0) [reply contrib/dlb/source/BUILD:19]: The error is still not found:
```
+ /build/.cache/bazel/_bazel_envoybuild/b570b5ccd0454dc9af9f65ab1833764d/sandbox/processwrapper-sandbox/1/execroot/envoy/bazel-out/k8-opt-exec-2B5CBBC6/bin/external/rules_foreign_cc/toolchains/make/bin/make libdlb.a PREFIX=/build/.cache/bazel/_bazel_envoybuild/b570b5ccd0454dc9af9f65ab1833764d/sandbox/processwrapper-sandbox/1/execroot/envoy/bazel-out/k8-fastbuild/bin/contrib/dlb/source/dlb
Compiling build/dlb.o
In file included from dlb.c:25:
dlb_priv.h:15:10: fatal error: dlb2_user.h: No such file or directory
   15 | #include "dlb2_user.h"
      |          ^~~~~~~~~~~~~
compilation terminated.
make: *** [Makefile:60: build/dlb.o] Error 1
_____ END BUILD LOGS _____
rules_foreign_cc: Build wrapper script location: bazel-out/k8-fastbuild/bin/contrib/dlb/source/dlb_foreign_cc/wrapper_build_script.sh
rules_foreign_cc: Build script location: bazel-out/k8-fastbuild/bin/contrib/dlb/source/dlb_foreign_cc/build_script.sh
rules_foreign_cc: Build log location: bazel-out/k8-fastbuild/bin/contrib/dlb/source/dlb_foreign_cc/Make.log

Target //contrib/dlb/source:dlb failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 0.843s, Critical Path: 0.19s
INFO: 2 processes: 2 internal.
FAILED: Build did NOT complete successfully
```

Comment (daixiang0) [reply contrib/dlb/source/BUILD:19]: I guess `filegroup` keep the dir layout so that Make cannot find it.

Comment (phlax) [reply contrib/dlb/source/BUILD:19]: probably so - wondering what layout is expected

see here for how to mangle file paths in a filegroup https://groups.google.com/g/bazel-discuss/c/x0nyiuQ5Q44/m/r7rGSQTFHAAJ

Comment (daixiang0) [reply contrib/dlb/source/BUILD:19]: It would raise some error:
```
ERROR: /build/.cache/bazel/_bazel_envoybuild/b570b5ccd0454dc9af9f65ab1833764d/external/intel_dlb/BUILD.bazel:18:9: unclosed string literal
ERROR: /build/.cache/bazel/_bazel_envoybuild/b570b5ccd0454dc9af9f65ab1833764d/external/intel_dlb/BUILD.bazel:19:1: unclosed string literal
ERROR: /build/.cache/bazel/_bazel_envoybuild/b570b5ccd0454dc9af9f65ab1833764d/external/intel_dlb/BUILD.bazel:19:1: Implicit string concatenation is forbidden, use the + operator
```

The generated:
```
genrule(
  name = 'copy_the_files',
  srcs = files,
  outs = [f.replace('the_directory', 'another_place/folder') for f in files],
  cmd = '
'.join(["mkdir -p $$(dirname $(location %s)) && cp $(location %s) $(location :%s)" % (f, f, f.replace("the_directory", "another_place/folder")) for f in files]),
)
```

Comment (daixiang0) [reply contrib/dlb/source/BUILD:19]: I simplify the copy:
```
filegroup(
    name = "header",
    srcs = glob(["**/dlb2_user.h"]),
    visibility = ["@envoy//contrib/dlb/source:__pkg__"],
)

genrule(
  name = "copy",
  srcs = [":header"],
  outs = ["dlb/libdlb"],
  cmd = "set -x;cp -a $(SRCS) $(OUTS)",
)
```
But it turns out:
```
cp -a external/intel_dlb/dlb/driver/dlb2/uapi/linux/dlb2_user.h bazel-out/k8-fastbuild/bin/external/intel_dlb/dlb/libdlb
```

@phlax as I show it in the slack, the original layout is like:
```
dlb
├── driver
│   └── dlb2
│       ├── user.h
└── libdlb
```

what I want is like:
```
dlb     
└── libdlb
        └── dlb2│
```

The `genrule` cannot operate the "external" dir since it is remote for `genrule`.

Comment (daixiang0) [reply contrib/dlb/source/BUILD:19]: @phlax do you have a good way to do it?

**Comment by daixiang0** [comment]
The CI failure is not related.

File: bazel/repositories.bzl [hunk 6]
```diff
@@ -1372,12 +1372,13 @@ def _intel_dlb():
...
-    srcs = glob([
-        "dlb/libdlb/**",
-    ]),
+    srcs = glob(["dlb/libdlb/*"]),
     visibility = ["@envoy//contrib/dlb/source:__pkg__"],
 )
 """,
+        patch_args = ["-p1"],
+        patches = ["@envoy//bazel/foreign_cc:dlb.patch"],
+        patch_cmds = ["cp dlb/driver/dlb2/uapi/linux/dlb2_user.h dlb/libdlb/"],
```
Comment (adisuissa) [inline bazel/repositories.bzl:1381]: nit: consider using a symbolic link (`ln`) instead of copying.

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!
I'll let @phlax have the final say.

File: bazel/repositories.bzl (hunk 6 above)
Comment (phlax) [reply bazel/repositories.bzl:1381]: i would stick with copy -  bazel can be a bit unpredictable with symbolic links
